File: golang/client/client_test.go
package client

import (
	"os"
	"testing"

	util "github.com/alibabacloud-go/tea-utils/service"
	"github.com/alibabacloud-go/tea/tea"
	"github.com/alibabacloud-go/tea/utils"
)

func Test_Api(t *testing.T) {
	config := &Config{
		AppKey:    tea.String(os.Getenv("IOT_APP_KEY")),
		AppSecret: tea.String(os.Getenv("IOT_APP_SECRET")),
		Domain:    tea.String("api.link.aliyun.com"),
	}
	client, err := NewClient(config)
	utils.AssertNil(t, err)

	params := map[string]interface{}{
		"input": "test",
	}
	req := &CommonParams{
		ApiVer: tea.String("1.0.0"),
	}
	body := &IoTApiRequest{
		Params:  params,
		Request: req,
	}
	runtime := &util.RuntimeOptions{}
	resp, err := client.DoRequest(tea.String("/kit/debug/ping"), tea.String("HTTPS"), tea.String("POST"), nil, body, runtime)
	utils.AssertNil(t, err)

	obj, err := util.ReadAsJSON(resp.Body)
	utils.AssertNil(t, err)

	m := util.AssertAsMap(obj)
	utils.AssertContains(t, "test", m["data"].(string))
}

File: java/src/test/java/com/aliyun/iotapigateway/ClientTest.java
package com.aliyun.iotapigateway;

import com.aliyun.iotapigateway.models.CommonParams;
import com.aliyun.iotapigateway.models.Config;
import com.aliyun.iotapigateway.models.IoTApiRequest;
import com.aliyun.tea.TeaResponse;
import com.aliyun.teautil.Common;
import com.aliyun.teautil.models.RuntimeOptions;
import org.junit.Assert;
import org.junit.Test;

import java.util.HashMap;
import java.util.Map;

public class ClientTest {

    @Test
    public void apiTest() throws Exception{
        Config config = new Config();
        config.appKey = System.getenv("IOT_APP_KEY");
        config.appSecret = System.getenv("IOT_APP_SECRET");
        config.domain = "api.link.aliyun.com";

        Client client = new Client(config);

        Map<String, Object> params = new HashMap<>();
        params.put("input", "test");

        CommonParams req = new CommonParams();
        req.apiVer = "1.0.0";

        IoTApiRequest body = new IoTApiRequest();
        body.params = params;
        body.request = req;

        RuntimeOptions runtime = new RuntimeOptions();

        TeaResponse teaResponse = client.doRequest("/kit/debug/ping", "HTTPS",
                "POST", null, body, runtime);

        Object obj = Common.readAsJSON(teaResponse.body);

        Map<String, Object> result = Common.assertAsMap(obj);
        Assert.assertEquals("test", result.get("data"));
    }
}
